pkg/controller/keystone/keystone_controller.go

```diff
@@ -120,12 +120,26 @@ func (r *ReconcileKeystone) Reconcile(request reconcile.Request) (reconcile.Resu
 		return reconcile.Result{}, nil
 	}
 
+	keystoneService := newKeystoneService(keystone)
+	_, err := controllerutil.CreateOrUpdate(context.Background(), r.client, keystoneService, func() error { return nil })
+
 	keystonePods, err := r.listKeystonePods(keystone.Name)
 	if err != nil {
 		return reconcile.Result{}, fmt.Errorf("failed to list command pods: %v", err)
 	}
 
-	if err := r.ensureCertificatesExist(keystone, keystonePods); err != nil {
+	keystoneServiceList := &core.ServiceList{}
+	labelSelector := labels.SelectorFromSet(map[string]string{"service": keystone.Name})
+	listOpts := client.ListOptions{LabelSelector: labelSelector}
+	if err := r.client.List(context.TODO(), keystoneServiceList, &listOpts); err != nil {
+		return reconcile.Result{}, err
+	}
+	var keystoneClusterIP string
+	if len(keystoneServiceList.Items) > 0 {
+		keystoneClusterIP = keystoneServiceList.Items[0].Spec.ClusterIP
```

Comment:
moreover I think that we should avoid situation when `keystoneClusterIP` because in that case we will create a certificates without serviceIP and it won't be regenerated, so I would exit here if it is empty and wait for it